sys/behave/src/Expand.hs

```diff
@@ -137,9 +152,13 @@ expand chsets (BNbexpr we (ProcInst procid chans vexps))  =  do
      case Map.lookup procid (procDefs tdefs) of
        Just (ProcDef chids vids bexp)
          -> do let chanmap = Map.fromList (zip chids chans)
-               wals  <- mapM (Eval.eval . subst (Map.map cstrConst we) (funcDefs tdefs) ) vexps
-               let we' = Map.fromList (zip vids wals)
-               expand chsets $ BNbexpr we' (relabel chanmap bexp)
+               let wals = map (TxsDefs.eval . subst (Map.map cstrConst we) (funcDefs tdefs) ) vexps
+               case Data.Either.partitionEithers wals of
```

Comment:
I'm thinking that, if we want to make this code transferable to anybody maybe it is a good idea to add some comments about what's going on here. Why are we partitioning the `Either`'s? What does each case represent? (for instance what does it mean to have no lefts?)

Reply:
Isn't that clear from what Eval does?